pyc64/basic.py

```diff
@@ -405,6 +412,51 @@ def execute_goto(self, cmd):
             if line not in self.program:
                 raise BasicError("undef'd statement")
             raise GotoLineException(self.program_lines.index(line))
+    """
+    on <index-1-based> goto|gosub <line1>,<line2> 
+
+    if index evaluate to 1 the execution proceed on line1
+    """            
+    def execute_on_goto_gosub(self,cmd):
+        gosub=False
+        if cmd.startswith("on"):
+            cmd=cmd[2:]
+        if cmd.find("goto")==-1 and cmd.find("gosub")==-1:
+            raise BasicError("syntax")
+        else:
+            # Find out index list
+            l2=(cmd[(cmd.find("go")+2):]).strip()
+            # DEBUG print(l2)
+            if l2.startswith("to"):
+                # goto branch
+                gosub=False
+                targetLineList=l2[2:]
+            elif l2.startswith("sub"):
+                gosub=True
+                targetLineList=l2[3:]
+            else:
+                raise BasicError("syntax")
+        targetLineList=targetLineList.strip()
+        # Arrayz...ize it        
+        lineTargetTuple=eval("("+targetLineList+")")
+        goInx=cmd.find("go")
+        expr=cmd[0:goInx]
+        print("lineTargetTuple= %s expr=  %s " % (lineTargetTuple, str(expr)))
```

Comment:
looks like a debug statement that should be removed

Reply:
Ok I will fix it